### PR TITLE
Centralize thumbnail viewing

### PR DESCRIPTION
**Pull request by EricClaeys**
* Replace the `videos/index.php`, `keograms/index.php`, and `startrails/index.php` with `show_thumbnails.php`.
* Improve the look of the thumbnails pages slightly.

### CODE REVIEW

**Comment by thomasjacquin** [comment]
@EricClaeys Does this fix issue #107 ?

Also, I think we need to make sure the code is not favouring people hosting the website locally on their RPi over people who host it on a web host.
For example, web hosts don't always have a lot of image and video editing programs installed. In my case, ffmpeg isn't installed.

image

**Comment by EricClaeys** [comment]
@thomasjacquin    Thomas,  I'll have to check the code.  It should complain about a 0-length file, then go to the next one.
For the 1-column file, is that a keogram?
Was your screenshot above from your Pi or a remote server?

I think we have remote servers covered.
We check for the ability to run "exec()" for the `ffmpeg` command, since many remote servers either don't have `ffmpeg` or "exec()" is disabled.  And we check that the PHP function to create thumbnails from .jpg/.png files is enabled.
**Did you find something else on a remote server that didn't work?**

One of my "to do" items is to add an option to create the thumbnails on the Pi and upload to the remote server.

**Comment by thomasjacquin** [comment]
@EricClaeys The screenshot is from my remote server. `exec()` works, yet `ffmpeg` is not installed. The creation of jpg/png thumbnails is functional.

Regarding issue #107, I noticed that the directory is called `keograms` but the individual image prefix is `keogram` which breaks the thumbnail naming.

Yes, the 1px wide image is a keogram. It happens every year around summer solstice at my location.

**Comment by EricClaeys** [comment]
@thomasjacquin   This PR fixes the issue with "keogram" versus "keograms" and the data not showing correctly.

I'd like to fix the problem with `ffmpeg` not being installed in a separate PR if that's ok with you, once this PR is merged.  If it is, is this PR ready for approval?

Not sure what to do with a 1-pixel wide image.  Can you post it so I can test?

**Comment by thomasjacquin** [comment]
@EricClaeys I agree, let's handle the `ffmpeg` issue in a different PR.

Here's an image that's 1px wide. https://www.thomasjacquin.com/allsky/keograms/keogram-20210626.jpg

File: index.php [hunk 1]
```diff
@@ -53,12 +53,13 @@
 
 	<ul id="sidebar" class="animated slideInLeft">
 		<li><i class="fa fa-2x fa-fw allsky-constellation" id="overlayBtn" title="Show constellations overlay" ng-click="toggleOverlay()" ng-class="{'active': showOverlay}"></i></li>
-		<li><a href="videos" title="Archived Timelapses"><i class="fa fa-2x fa-fw fa-play-circle"></i></a></li>
-		<li><a href="keograms" title="Archived Keograms"><i class="fa fa-2x fa-fw fa-barcode"></i></a></li>
-		<li><a href="startrails" title="Archived Startrails"><i class="fa fa-2x fa-fw fa-circle-notch"></i></a></li>
+		<li><a href="show_thumbnails.php?dir=videos&prefix=allsky&title=Timelapse%20Videos" title="Archived Timelapses"><i class="fa fa-2x fa-fw fa-play-circle"></i></a></li>
```
Comment (thomasjacquin) [inline index.php:56]: I do like the old url better. It is easier to remember `/videos`, `/startrails` and `/keograms` are easier to remember and nicer when sharing the URL.

I understand you've refactored to remove duplicated code but it there a way we keep the urls intact and call the unified code inside the respective index.php?

Comment (EricClaeys) [reply index.php]: I made the changes - added 3 */index.php files and modified index.php, functions.php, and show_thumbnails.php.

**Review by thomasjacquin — APPROVED** [review]
Nice, I think it's better with the human readable URLs. Thanks

**Comment by Pedritoprint** [comment]
@EricClaeys you remember my Problem one month ago? i have the same, because my WebServer don't understand the ffmpeg-command. "ffmpeg" is installed and the same version like RP has installed, but i think, the ffmpeg-configuration is different ant till today, i can't find a solution to make it running on my synology-NAS-Webserver and i'm still working on it.

**Comment by EricClaeys** [comment]
@Pedritoprint  Try the attached file.  Rename your current `functions.php` file to `functions.php-ORIGINAL` then install this file, removing the trailing `.txt`.
It won't solve your problem but may give a better idea of what's happening.
Press F12 in the browser, then click on `>>` then `Console`.  This file should produce some message.

[functions.php.txt](https://github.com/thomasjacquin/allsky-website/files/8626859/functions.php.txt)

**Comment by Pedritoprint** [comment]
This is the output in console. I think there is a newer WEB-SW on github?
greetings Peter
image

**Comment by EricClaeys** [comment]
@Pedritoprint. Peter, I suggest installing the newest code.  It looks like there are some new and some old files.  After you do that, then try the functions.php.txt file I posted.

**Comment by Pedritoprint** [comment]
here the new one with latest Web-SW.
image

**Comment by EricClaeys** [comment]
@Pedritoprint. What happens when you go to the timelapse video link?  Does it create the thumbnails or produce a Console message?

**Comment by Pedritoprint** [comment]
This is the result after press timelaps video link. No thumbnails generated, but link to videos ar working.
image

**Pull request merged**